### DISCUSSION TITLE
Unable to create an index due to 15m limit, even with OLAP mode

### DISCUSSION THREAD

**Discussion by chrisordio** [planetscale/discussion#645] [Q&A]
I have a rather large table with about 68 million records.  creating a new index to improve performance is taking longer than 15 minutes.  I tried to get around by using OLAP, but this is not working.  I still get an error.  Please see the sample below.

```
mysql> SET workload=OLAP;
Query OK, 0 rows affected (0.10 sec)

mysql> CREATE INDEX <blah blah blah>
ERROR 1105 (HY000): unknown: target: ordin.-.primary: vttablet: rpc error: code = Canceled desc = (errno 2013) due to context deadline exceeded, elapsed time: 15m0.000514006s, killing query ID 1043 (CallerID: kt6t46pvb128kderfk9l)```

**Comment by samlambert** [comment]
You should turn safe migrations on and do this through a deploy request.

**Reply by chrisordio** [reply]
im new to planetscale so forgive me for the dumb question.  so i want to test this schema change on this branch that is outside of main.  the dropdown next to the "Create Deploy Request" feature has only one option and that is production.  This isn't what i want to do.

How can I apply this change to the branch and not main? this may be important information.  I created a branch from a backup and chose prod as the base branch.  now I'm thinking maybe i need to create a branch from this branch, without data, and then apply the schema change that way?

**Reply by mscoutermarsh** [reply]
Hi @chrisordio,

This is what I'd recommend. Create a new branch off of `main`. For this example, let's call it `dev`. This branch won't have any data in it.

Connect to the `dev` branch (or use the web console) and add your index. Then, open up a Deploy Request from `dev` into `main`. This will allow you to apply the index to `main`.

Once it's complete, you will have a 30 minute revert period. You can check directly in `main` if it has had the desired impact on your query performance. If it doesn't, you can revert the deploy request.

**Reply by chrisordio** [reply]
hi @mscoutermarsh thanks for the help.  i did something similar.  i created a `dev` branch but seeded it with data from prod.  i then created a branch from `dev` with no data. added an index there and created a deploy request.

the reason i need data in `dev` is that i needed to test out new indexes and their affect on improving performance.  in any case, my deploy request is taking much longer than expected. 3% in 4 hours

**Reply by mscoutermarsh** [reply]
Thanks, I replied over in the other thread.
